File: src/main/java/net/obesephoenix/koalakraftcrystals/events/onEntityDamage.java
package net.obesephoenix.koalakraftcrystals.events;

import net.obesephoenix.koalakraftcrystals.KoalaKraftCrystals;
import org.bukkit.Bukkit;
import org.bukkit.NamespacedKey;
import org.bukkit.entity.Item;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.EntityDamageEvent;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;
import org.bukkit.persistence.PersistentDataType;
import org.bukkit.util.Vector;

import javax.naming.Name;

public class onEntityDamage implements Listener {

    @EventHandler
    public void onEvent(EntityDamageEvent event) {
        if (!(event.getEntity() instanceof Item)) return;
        Item entity = (Item) event.getEntity();
        ItemStack item = entity.getItemStack();
        ItemMeta meta = item.getItemMeta();
        if (meta == null) return;

        if (meta.getPersistentDataContainer().has(new NamespacedKey(KoalaKraftCrystals.instance, "crystal_id")
                , PersistentDataType.STRING)) {
            event.setCancelled(true);
        }
    }
}

File: src/main/kotlin/net/obesephoenix/koalakraftcrystals/events/OnEntityDamage.kt
package net.obesephoenix.koalakraftcrystals.events

import org.bukkit.event.entity.EntityDamageEvent
import org.bukkit.inventory.meta.ItemMeta
import org.bukkit.NamespacedKey
import net.obesephoenix.koalakraftcrystals.KoalaKraftCrystals
import org.bukkit.entity.Item
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.persistence.PersistentDataType

class OnEntityDamage : Listener {
    @EventHandler
    fun onEvent(event: EntityDamageEvent) {
        if (event.entity !is Item) return
        val entity = event.entity as Item
        val item = entity.itemStack
        val meta = item.itemMeta ?: return
        if (meta.persistentDataContainer.has(
                NamespacedKey(KoalaKraftCrystals.instance, "crystal_id"), PersistentDataType.STRING
            )
        ) {
            event.isCancelled = true
        }
    }
}